internal/tui/components/dialogs/restart.go

```diff
@@ -0,0 +1,327 @@
+			r.loadItems(),
+			r.list.SetSize(r.listWidth(), r.listHeight()),
+		)
+	case tea.KeyPressMsg:
+		switch {
+		case key.Matches(msg, r.keyMap.Select):
+			selectedItem := r.list.SelectedItem()
+			if selectedItem == nil {
+				return r, nil
+			}
+			item := (*selectedItem).Value()
+			if item.Disabled {
+				return r, nil // Can't restart disabled items
+			}
+
+			if r.dialogType == "mcp" {
+				return r, tea.Sequence(
+					util.CmdHandler(CloseDialogMsg{}),
+					util.CmdHandler(RestartMCPMsg{Name: item.Name}),
+				)
+			} else {
+				return r, tea.Sequence(
+					util.CmdHandler(CloseDialogMsg{}),
+					util.CmdHandler(RestartLSPMsg{Name: item.Name}),
+				)
+			}
+		case key.Matches(msg, r.keyMap.Close):
+			return r, util.CmdHandler(CloseDialogMsg{})
+		default:
+			u, cmd := r.list.Update(msg)
+			r.list = u.(list.FilterableList[list.CompletionItem[RestartItem]])
+			return r, cmd
+		}
+	}
+	return r, nil
+}
+
+func (r *restartDialogCmp) View() string {
+	t := styles.CurrentTheme()
+	title := "Restart MCP"
+	if r.dialogType == "lsp" {
+		title = "Restart LSP"
+	}
+
+	header := t.S().Base.Padding(0, 1, 1, 1).Render(core.Title(title, r.width-4))
+	content := lipgloss.JoinVertical(
+		lipgloss.Left,
+		header,
+		r.list.View(),
+		"",
+		t.S().Base.Width(r.width-2).PaddingLeft(1).AlignHorizontal(lipgloss.Left).Render(r.help.View(r.keyMap)),
+	)
+	return r.style().Render(content)
+}
+
+func (r *restartDialogCmp) Cursor() *tea.Cursor {
+	if cursor, ok := r.list.(util.Cursor); ok {
+		cursor := cursor.Cursor()
+		if cursor != nil {
+			cursor = r.moveCursor(cursor)
+		}
+		return cursor
+	}
+	return nil
+}
+
+func (r *restartDialogCmp) loadItems() tea.Cmd {
+	var items []RestartItem
+
+	if r.dialogType == "mcp" {
+		// Load MCP items
+		cfg := config.Get()
+		mcpStates := agent.GetMCPStates()
+
+		for name, mcpConfig := range cfg.MCP {
+			state := "unknown"
+			disabled := false
+			description := fmt.Sprintf("Type: %s", mcpConfig.Type)
+
+			if mcpInfo, exists := mcpStates[name]; exists {
+				state = mcpInfo.State.String()
+				if mcpInfo.ToolCount > 0 {
+					description += fmt.Sprintf(", Tools: %d", mcpInfo.ToolCount)
+				}
+				if mcpInfo.Error != nil {
+					description += fmt.Sprintf(", Error: %s", mcpInfo.Error.Error())
+				}
+			}
+
+			// SSE MCPs cannot be restarted
+			if mcpConfig.Type == config.MCPSse {
+				disabled = true
+				description += " (cannot restart SSE)"
+			}
+
+			items = append(items, RestartItem{
+				Name:        name,
+				Type:        string(mcpConfig.Type),
+				State:       state,
+				Description: description,
+				Disabled:    disabled,
+			})
+		}
+	} else {
+		// Load LSP items
+		cfg := config.Get()
+		// We need to get LSP states from the app, but we can't import app here
+		// For now, let's create a basic list from configuration
+		for name := range cfg.LSP {
+			items = append(items, RestartItem{
+				Name:        name,
+				Type:        "lsp",
+				State:       "unknown",
```

Comment:
I think this needs to implement the state stuff here?

Reply:
<img width="602" height="240" alt="Screenshot 2025-08-26 at 06 42 09" src="https://github.com/user-attachments/assets/23971e4c-722b-44b1-ab97-722d32916db1" />